nestkernel/ntree_impl.h

```diff
@@ -478,7 +481,11 @@ Ntree< D, T, max_capacity, max_depth >::insert( Position< D > pos, const T& node
   if ( leaf_ )
   {
 
-    assert( ( pos >= lower_left_ ) && ( pos < lower_left_ + extent_ ) );
+    for ( int i = 0; i < D; ++i )
+    {
+      assert( ( pos - lower_left_ )[ i ] > -std::numeric_limits< double >::epsilon()
```

Comment:
Could you add a comment about why we need the negative limit here?

Reply:
Done!